components/Form/Edit.js

```diff
@@ -0,0 +1,108 @@
+import { useState, useEffect } from "react";
+import { FormContainer, FormButton, AddButton } from "./styles-form";
+import { Heading } from "./styles-form";
+
+export default function EditRecipe({ id, onSubmit }) {
+  const initialIngredients = [{ name: "", quantity: "" }];
+
+  const [recipe, setRecipe] = useState({
+    name: "",
+    ingredients: initialIngredients,
+    quantity: 0,
+    steps: "",
+    timers: "",
+  });
+
+  useEffect(() => {
+    if (id) {
+      fetchRecipe(id);
+    }
+  }, [id]);
+
```

Comment:
I would also recommend you put the fetchRecipe out of the component and put it into the editpage[id].js

If I understood Roland correctly, that is also what he meant.